framework/DialogPanel/DialogPanel.bbj

```diff
@@ -65,44 +68,44 @@ class public DialogPanel extends BBjWidget implements Icon
         
     methodend
 
-    method public void renderTopBar(BBjChildWindow modalWnd!)
-       #Header! = modalWnd!.addChildWindow(modalWnd!.getAvailableControlID(),0,0,0,0,"",$00108800$,BBjAPI().getSysGui().getAvailableContext())
-       #Header!.addStyle("headerWnd")
-       #Header!.addPanelStyle("headerPanel")
+    method public void renderTopBar()
+       #header! = #dialogWnd!.addChildWindow(#dialogWnd!.getAvailableControlID(),0,0,0,0,"",$00108800$,BBjAPI().getSysGui().getAvailableContext())
+       #header!.addStyle("headerWnd")
+       #header!.addPanelStyle("headerPanel")
 
-       #LblTitle! = #Header!.addStaticText(#Header!.getAvailableControlID(),0,0,0,0, #getText())
-       #LblTitle!.addStyle("headerTitle")
+       #lblTitle! = #header!.addStaticText(#header!.getAvailableControlID(),0,0,0,0, #getText())
+       #lblTitle!.addStyle("headerTitle")
        
+       #closeIconWnd! = #header!.addChildWindow(#header!.getAvailableControlID(), 0,0,0,0,"",$00100800$,BBjAPI().getSysGui().getAvailableContext())
+       #closeIconWnd!.addStyle("closeIconWnd")
+       iconpool$=stbl("icon-pool", err=*next)
+       ico$="x"
+       if iconpool$="fa" then ico$="fas-window-close"
```

Comment:
just a suggestion for better scalability - 
Can we create a mapping file for icon names, may be in a json or similar format, so that we can avoid these if..then blocks for providing default icon values? Something like - 
```
'iconName': 'close',
'mappings':{
  'fa' : 'fas-window-close',
  'feather': 'x',
  ...
}
```
Then we can use this mappings to render the icons specific to a certain pool by just using the generic `iconName`(in this case `close`) in our code. Because if you think about it wherever we are using `#setIcon` we need to add a if..then block to provide a default icon name.

Reply:
I'd leave the commented code inside the showcase for now, which is still highly WIP and will receive more attention hopefully soon. 

I agree on the icon file mapping. I've created https://github.com/BBj-Plugins/WebKit/issues/91